lib/ansible/modules/windows/win_firewall_rule.ps1

```diff
@@ -287,167 +178,99 @@ $remoteip = Get-AnsibleParam -obj $params -name "remoteip" -type "str" -default
-} else {
-    $result.fwsettings.Add("Profiles", $(ConvertTo-TitleCase($profiles)))
-}
-
-$result.fwsettings.Add("Edge traversal", $(ConvertTo-TitleCase($edge)))
-
-if ($interfacetypes -ne $null) {
-    $result.fwsettings.Add("InterfaceTypes", $(ConvertTo-TitleCase($interfacetypes)))
-}
-
-switch($security) {
-    "Authenticate" { $security = "Authenticate" }
-    "AuthDynEnc" { $security = "AuthDynEnc" }
-    "AuthEnc" { $security = "AuthEnc" }
-    "AuthNoEncap" { $security = "AuthNoEncap" }
-    "NotRequired" { $security = "NotRequired" }
-}
-$result.fwsettings.Add("Security", $security)
-
-# FIXME: Define unsupported options
-#$result.fwsettings.Add("Grouping", "")
-#$result.fwsettings.Add("Rule source", "Local Setting")
-
-$get = getFirewallRule($result.fwsettings)
-$result.msg += $get.msg
-
-if ($get.failed) {
-    $result.error = $get.error
-    $result.output = $get.output
-    Fail-Json $result $result.msg
-}
-
-$result.diff = $get.diff
-
-if ($state -eq "present") {
-    if (-not $get.exists) {
+    if ($existingRule -is [System.Array]) {
+        Fail-Json $result "Multiple firewall rules with name '$name' found."
+    }
 
-        $create = createFireWallRule($result.fwsettings)
-        $result.msg += $create.msg
-        $result.diff = $create.diff
+    $rule = New-FWRule -name $name `
+                       -description $description `
+                       -direction $direction `
+                       -action $action `
+                       -applicationName $program `
+                       -serviceName $service `
+                       -enabled $enabled `
+                       -profiles $profiles `
+                       -localAddresses $localip `
+                       -remoteAddresses $remoteip `
+                       -localPorts $localport `
+                       -remotePorts $remoteport `
+                       -protocol $protocol `
+                       -interfaceTypes $interfacetypes `
+                       -edgeTraversalOptions $edge `
+                       -secureFlags $security
+
+    $fwPropertiesToCompare = @('Name','Description','Direction','Action','ApplicationName','ServiceName','Enabled','Profiles','LocalAddresses','RemoteAddresses','LocalPorts','RemotePorts','Protocol','InterfaceTypes', 'EdgeTraversalOptions', 'SecureFlags')
+
+    if ($state -eq "absent") {
+        if ($existingRule -eq $null) {
+            $result.msg = "Firewall rule '$name' does not exist."
+        } else {
+            if ($diff_support) {
+                foreach ($prop in $fwPropertiesToCompare) {
+                    $result.diff.prepared += "-[$($prop)='$($existingRule.$prop)']`n"
+                }
+            }
 
-        if ($create.failed) {
-            $result.error = $create.error
-            $result.output = $create.output
-            Fail-Json $result $result.msg
+            if (-not $check_mode) {
+                $fw.Rules.Remove($existingRule.Name)
```

Comment:
What is the error message like if this goes wrong, should we be catching it and calling `Fail-Json` with a better error message instead?